core/camel-api/src/main/java/org/apache/camel/spi/ModelReifierFactory.java

```diff
@@ -28,22 +28,54 @@
  */
 public interface ModelReifierFactory {
 
+    /**
+     * Service factory key for custom factories.
+     */
+    String FACTORY = "model-reifier-factory";
```

Comment:
@lburgazzoli so a custom JAR with a class that extends the default factory and has @JdkService(FACTORY) as annotation should make camel detect your class instead of the default